synthesizer/src/vm/helpers/cost.rs

```diff
@@ -98,58 +117,143 @@ pub fn execution_cost<N: Network, C: ConsensusStorage<N>>(
-            PlaintextType::Literal(..) => Ok(100_000),
-            plaintext_type => bail!("`hash.psd4` is not supported for plaintext type '{plaintext_type}'"),
-        },
-        Command::Instruction(Instruction::HashPSD8(hash)) => match hash.destination_type() {
-            PlaintextType::Literal(LiteralType::Address) | PlaintextType::Literal(LiteralType::Group) => Ok(800_000),
-            PlaintextType::Literal(..) => Ok(200_000),
-            plaintext_type => bail!("`hash.psd8` is not supported for plaintext type '{plaintext_type}'"),
-        },
-        Command::Instruction(Instruction::HashSha3_256(_)) => Ok(100_000),
-        Command::Instruction(Instruction::HashSha3_384(_)) => Ok(100_000),
-        Command::Instruction(Instruction::HashSha3_512(_)) => Ok(100_000),
+        Command::Instruction(Instruction::Cast(cast)) => {
+            let cast_type = cast.cast_type();
+            match cast_type {
+                CastType::Plaintext(PlaintextType::Literal(_)) => Ok(500),
+                CastType::Plaintext(plaintext_type) => Ok(plaintext_size_in_bytes(stack, plaintext_type)?
+                    .saturating_mul(CAST_PER_BYTE_COST)
+                    .saturating_add(CAST_COMMAND_BASE_COST)),
+                _ => Ok(500),
+            }
+        }
+        Command::Instruction(Instruction::CastLossy(cast_lossy)) => {
+            let cast_type = cast_lossy.cast_type();
+            match cast_type {
+                CastType::Plaintext(PlaintextType::Literal(_)) => Ok(500),
+                CastType::Plaintext(plaintext_type) => Ok(plaintext_size_in_bytes(stack, plaintext_type)?
+                    .saturating_mul(CAST_PER_BYTE_COST)
+                    .saturating_add(CAST_COMMAND_BASE_COST)),
+                _ => Ok(500),
+            }
+        }
+        Command::Instruction(Instruction::CommitBHP256(commit)) => {
+            size_cost(commit.operands(), HASH_BHP_PER_BYTE_COST, HASH_BHP_BASE_COST)
+        }
+        Command::Instruction(Instruction::CommitBHP512(commit)) => {
+            size_cost(commit.operands(), HASH_BHP_PER_BYTE_COST, HASH_BHP_BASE_COST)
+        }
+        Command::Instruction(Instruction::CommitBHP768(commit)) => {
+            size_cost(commit.operands(), HASH_BHP_PER_BYTE_COST, HASH_BHP_BASE_COST)
+        }
+        Command::Instruction(Instruction::CommitBHP1024(commit)) => {
+            size_cost(commit.operands(), HASH_BHP_PER_BYTE_COST, HASH_BHP_BASE_COST)
+        }
+        Command::Instruction(Instruction::CommitPED64(commit)) => {
+            size_cost(commit.operands(), HASH_PER_BYTE_COST, HASH_BHP_PER_BYTE_COST)
+        }
+        Command::Instruction(Instruction::CommitPED128(commit)) => {
+            size_cost(commit.operands(), HASH_PER_BYTE_COST, HASH_BHP_BASE_COST)
+        }
+        Command::Instruction(Instruction::Div(div)) => {
+            let operands = div.operands();
+            if operands.len() == 2 {
```

Comment:
I like to use `ensure!(operands.len() == 2, "div opcode must have exactly two operands.")`, so that we don't have to indent and for readability.

Reply:
Same for the other cases below

Reply:
Fine point sir